src/main/vcpf4/usbd_cdc_vcp.c

```diff
@@ -44,7 +44,7 @@ __IO uint32_t bDeviceState = UNCONNECTED; /* USB device status */
 
 /* This is the buffer for data received from the MCU to APP (i.e. MCU TX, APP RX) */
 extern uint8_t APP_Rx_Buffer[];
-extern uint32_t APP_Rx_ptr_out;
+extern volatile uint32_t APP_Rx_ptr_out;
```

Comment:
Good catch! Did this solve the problems of the new GCC compiler optimizations?

Reply:
Yes, according to my testing.